core/node/state_keeper/src/mempool_actor.rs

```diff
@@ -131,6 +348,14 @@ impl MempoolFetcher {
                 .await
                 .context("failed syncing mempool")?;
 
+            let (transactions_with_constraints, transactions_to_return) =
+                filter_unsafe_deposits(transactions_with_constraints, &mut storage).await?;
```

Comment:
The approach looks good! Just a couple of comments
- let's log tx hashes in `transactions_to_return`
- let's maybe add a config var `skip_filtering_unsafe_deposits` with default=false so it's possible to disable filtering in case it doesn't work as expected and filters some safe deposits